src/lightning_app/components/serve/auto_scaler.py

```diff
@@ -354,40 +316,20 @@ async def startup_event():
         def shutdown_event():
             fastapi_app.SEND_TASK.cancel()
 
-        def authenticate_private_endpoint(credentials: HTTPBasicCredentials = Depends(security)):
-            AUTO_SCALER_AUTH_PASSWORD = os.environ.get("AUTO_SCALER_AUTH_PASSWORD", "")
-            if len(AUTO_SCALER_AUTH_PASSWORD) == 0:
-                logger.warn(
-                    "You have not set a password for private endpoints! To set a password, add "
-                    "`--env AUTO_SCALER_AUTH_PASSWORD=<your pass>` to your lightning run command."
-                )
-            current_password_bytes = credentials.password.encode("utf8")
-            is_correct_password = secrets.compare_digest(
-                current_password_bytes, AUTO_SCALER_AUTH_PASSWORD.encode("utf8")
-            )
-            if not is_correct_password:
-                raise HTTPException(
-                    status_code=401,
-                    detail="Incorrect password",
-                    headers={"WWW-Authenticate": "Basic"},
-                )
-            return True
-
         @fastapi_app.get("/system/info", response_model=_SysInfo)
-        async def sys_info(authenticated: bool = Depends(authenticate_private_endpoint)):
+        async def sys_info():
             return _SysInfo(
-                num_workers=len(self._servers),
-                servers=self._servers,
+                num_workers=len(self.servers),
+                servers=self.servers,
                 num_requests=fastapi_app.num_current_requests,
                 processing_time=fastapi_app.last_processing_time,
                 global_request_count=fastapi_app.global_request_count,
             )
 
         @fastapi_app.put("/system/update-servers")
-        async def update_servers(servers: List[str], authenticated: bool = Depends(authenticate_private_endpoint)):
-            async with lock:
-                self._servers = servers
-            self._iter = cycle(self._servers)
+        async def update_servers(servers: List[str]):
```

Comment:
I guess the impact here is that users who don't set basic auth won't get a protected endpoint?

Reply:
So to summarise, user impact here is, users won't be able to set authentication for their autoscaler components using code changes. But they have much better option with the the new argument